### PR TITLE
New SCIM service

### PR DESCRIPTION
**Pull request by tauceti2**
SCIM sends information about users and group and memberships.

### CODE REVIEW

File: gen/scim [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+our $STATUS_VALID;                *STATUS_VALID =            \'VALID';
+our $STATUS_EXPIRED;              *STATUS_EXPIRED =          \'EXPIRED';
+our $STATUS_SUSPENDED;            *STATUS_SUSPENDED =        \'SUSPENDED';
+
+our $userStruc = {};
+our $u_status = {};
+our $u_email = {};
+our $u_name = {};
+our $u_login = {};
```
Comment (licehammer) [inline gen/scim]: Please use camelCase.
There is no need to initialize these variables as empty hash reference.
Do you really need these variable to be global?

File: gen/scim [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+use warnings;
+use perunServicesInit;
+use perunServicesUtils;
+use Perun::Agent;
+use Perun::GroupsAgent;
+use open qw/:std :utf8/;
+use JSON::XS;
+use utf8;
+
+local $::SERVICE_NAME = "scim";
```
Comment (licehammer) [inline gen/scim]: Are you sure about naming this service "scim"?  How would slave script for this service look like?

File: gen/scim [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        foreach my $groupData (($resourceData->getChildElements)[0]->getChildElements) {
+                my $groupMembersLogins = processGroups $groupData;
+        }
+}
+
+# PREPARE USERSDATA TO JSON
+my @users;
+foreach my $uid (sort keys %$userStruc) {
+        my $user = {};
+        $user->{"id"} = $uid;
```
Comment (licehammer) [inline gen/scim]: Do we want to reveal internal Perun ID to other systems?

File: gen/scim [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+                        $userStruc->{$uid}->{$u_status} = $status;
+                } elsif ($memberStatus eq $STATUS_SUSPENDED && $status eq $STATUS_VALID){
+                        # change from SUSPENDED to VALID
+                        $userStruc->{$uid}->{$u_status} = $status;
+                } elsif ($memberStatus eq $STATUS_SUSPENDED && $status eq $STATUS_EXPIRED){
+                        # change from SUSPENDED to EXPIRED
+                        $userStruc->{$uid}->{$u_status} = $status;
+                }
+        }
+        else{
```
Comment (licehammer) [inline gen/scim]: Missing space before '{'

File: gen/scim [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+                }
+        }
+}
+
+## creates structure for memberships
+sub processMemberships {
+        my ($gid, $uid) = @_;
+
+        unless(exists $membershipStruc->{$gid}->{$uid}) {
+                $membershipStruc->{$gid}->{$uid} = {};
```
Comment (licehammer) [inline gen/scim]: Is membershipStruc used for anything?

File: gen/scim (hunk 2 above)
Comment (tauceti2) [inline gen/scim]: The slave already exists. It just drop files at the target system and external script is usually getting those files for processing.

File: gen/scim (hunk 3 above)
Comment (tauceti2) [inline gen/scim]: actually this is the only way, how you can link memberships and users

File: gen/scim (hunk 5 above)
Comment (tauceti2) [inline gen/scim]: yes for printing users membership into the groups.scim

Comment (licehammer) [inline gen/scim]: My bad, I overlooked that part of code.

**Pull request merged**